tests/Microsoft.Identity.Test.Unit/CoreTests/InstanceTests/DTSAuthorityTest.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Instance;
+using Microsoft.Identity.Test.Common;
+using Microsoft.Identity.Test.Common.Core.Mocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.CoreTests.InstanceTests
+{
+    [TestClass]
+    public class DTSAuthorityTest
+    {
+        string _authorityUri = $"https://co2agg04-passive-dsts.dsts.core.azure-test.net/dstsv2/";
+
+        [TestMethod]
+        public void Validate_MinNumberOfSegments()
+        {
+            try
+            {
+                var instance = Authority.CreateAuthority(_authorityUri);
+
+                Assert.Fail("test should have failed");
+            }
+            catch (Exception exc)
+            {
+                Assert.IsInstanceOfType(exc, typeof(ArgumentException));
+                Assert.AreEqual(MsalErrorMessage.DstsAuthorityUriInvalidPath, exc.Message);
+            }

```

Comment:
Use `Assert.ThrowsException instead.

Reply:
Aeert.ThrowsException will not allow checking message.

Reply:
It returns the exception that was thrown. So something like:
```csharp
            var exception = Assert.ThrowsException<ArgumentException>(
               () => Authority.CreateAuthority(_authorityUri));
            Assert.AreEqual(MsalErrorMessage.DstsAuthorityUriInvalidPath, exception.Message);
```